noodles-bed/src/de.rs

```diff
@@ -83,28 +93,283 @@ where
+    //     V: Visitor<'de>,
+    // {
+    //     dbg!("string");
+    //     visitor.visit_str(self.input)
+    // }
+
+    fn deserialize_string<V: Visitor<'de>>(self, visitor: V) -> Result<V::Value> {
+        dbg!("string");
+        self.next_field().and_then(|f| visitor.visit_str(f.into()))
+    }
+
+    fn deserialize_identifier<V>(self, visitor: V) -> Result<V::Value>
+    where
+        V: Visitor<'de>,
+    {
+        dbg!("identifier");
+        self.deserialize_string(visitor)
+    }
+
+    fn deserialize_ignored_any<V>(self, visitor: V) -> Result<V::Value>
+    where
+        V: Visitor<'de>,
+    {
+        dbg!(self.input);
+        dbg!("deserialize_ignored_any");
+        drop(self.input);
+        // panic!(); // uncomment to stop loop
+        visitor.visit_unit()
+    }
+
+    // /// We can probably use forward_to_deserialize_any for many types, as bed Records are self-describing.
+    // forward_to_deserialize_any! {
+    //       bool i8 i16 i32 i64 i128 u8 u16 u32 u64 u128 f32 f64 char str string
+    //       bytes byte_buf option unit unit_struct newtype_struct tuple
+    //       tuple_struct map enum identifier ignored_any
+    // }
+
     forward_to_deserialize_any! {
-          bool i8 i16 i32 i64 i128 u8 u16 u32 u64 u128 f32 f64 char str string
+          bool i8 i16 i32 i64 i128 u8 u16 u32 u64 u128 f32 f64 char str
           bytes byte_buf option unit unit_struct newtype_struct tuple
-          tuple_struct map enum identifier ignored_any
+          tuple_struct enum
     }
 }
 
 /// SeqAccess will be needed to deserialize sequences and structs.
-impl<'de, I> SeqAccess<'de> for Record3Deserializer<I>
-where
-    I: Iterator<Item = io::Result<Record<3>>>,
-{
+impl<'de> SeqAccess<'de> for RecordDeserializer<'de> {
     type Error = Error;
 
     fn next_element_seed<T>(&mut self, seed: T) -> Result<Option<T::Value>>
     where
         T: DeserializeSeed<'de>,
     {
-        todo!()
+        seed.deserialize(&mut *self).map(Some)
+    }
+
+    // fn next_element<T>(&mut self) -> Result<Option<T>>
+    // where
+    //     T: Deserialize<'de>,
+    // {
+    //     self.next_element_seed(std::marker::PhantomData)
+    // }
+
+    fn size_hint(&self) -> Option<usize> {
+        None
+    }
+}
+
+impl<'de, 'a> MapAccess<'de> for RecordDeserializer<'de> {
+    type Error = Error;
+
+    fn next_key_seed<K>(&mut self, seed: K) -> Result<Option<K::Value>>
+    where
+        K: DeserializeSeed<'de>,
+    {
+        // // Check if there are no more entries.
+        // if self.de.peek_char()? == '}' {
+        //     return Ok(None);
+        // }
+        // // Comma is required before every entry except the first.
+        // if !self.first && self.de.next_char()? != ',' {
+        //     return Err(Error::ExpectedMapComma);
+        // }
+        // self.first = false;
+        // // Deserialize a map key.
+        // seed.deserialize(&mut *self.de).map(Some)
+        seed.deserialize(&mut *self).map(Some)
```

Comment:
It seems to be getting stuck here, as this function is called over and over again, without returning `Ok(None)`. [`next_key_seed`](https://docs.rs/serde/latest/serde/de/trait.MapAccess.html#tymethod.next_key_seed) should return `Ok(None)` when there are no more entries left.